utils.go

```diff
@@ -0,0 +1,49 @@
+package main
+
+import (
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+)
+
+func execute(in io.Reader, n string, args ...string) (string, int, error) {
+	cmd := exec.Command(n, args...)
+	cmd.Stdin = in
+	log.Printf("executing: %s", cmd)
+	b, err := cmd.CombinedOutput()
+	if err != nil {
+		eerr, ok := err.(*exec.ExitError)
+		if ok && eerr != nil {
+			return string(b), eerr.ExitCode(), eerr
+		}
+		return "", -1, err
+	}
+
+	return string(b), 0, nil
+}
+
+func isInstanceAlreadyRunning(n string) (bool, error) {
```

Comment:
It is not obvious why a sting is needed as an argument. Comment?

Reply:
Added comment. I'm still trying to find a way to determine if an instance is already running that doesn't involve calling `ps`.